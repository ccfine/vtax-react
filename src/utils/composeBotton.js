/*
 * @Author: liuchunxiu 
 * @Date: 2018-05-16 14:51:15 
 * @Last Modified by: liuchunxiu
 * @Last Modified time: 2018-07-19 17:21:14
 */
import React from "react";
/*import ButtonWithPut from "../compoments/buttonWithPut";*/
import ButtonReset from 'compoments/buttonReset'
import SubmitOrRecall from "compoments/buttonModalWithForm/SubmitOrRecall.r";
import PermissibleRender from "compoments/permissible/PermissibleRender.r";
import ButtonMarkModal from 'compoments/buttonMarkModal'
import ButtonConsistent from 'compoments/buttonConsistent'
import {FileExport,FileImportModal,AutoFileUpload,FileUndoImportModal,FileDownload} from 'compoments';
import ButtonTableAction from 'compoments/buttonTableAction'
import ButtonSwitch from 'compoments/buttonSwitch'
import {ButtonModalWithForm} from 'compoments'
//1：暂存 2：提交
// 判断当前状态是否提交
const isDisabled = (statusParam = {})=> parseInt(statusParam.status, 0) === 2;
const formatJson = (param) => {
    if(("authMonth" in param)){
        param['taxMonth'] = param['authMonth'];
    }
    return param
}

//新增
const getConsistentOptions = (item, statusParam) => {
    if(("selectedRowKeys" in item)){
        return {
            ...item,
            disabled: isDisabled(statusParam) ? isDisabled(statusParam) : item.selectedRowKeys && !item.selectedRowKeys.length>0,
            style: item.style || {marginRight:5},
        };
    }else{
        return {
            ...item,
            disabled: isDisabled(statusParam),
            style: item.style || {marginRight:5},
        };
    }
};

// 重算参数
const getResetOptions = (item, statusParam) => {
    // url,参数不存在，或是已经提交 重算不可用
    return {
        ...item,
        url:item.url,
        filters:formatJson(item.params),
        disabled: isDisabled(statusParam),
        onSuccess: item.onSuccess,
        style: { marginRight: 5 }
    };
};

// 提交参数
const getSubmitOptions = (item, statusParam) => {
    return {
        ...item,
        url: item.url,
        disabled: isDisabled(statusParam),
        onSuccess: item.onSuccess,
        initialValue: formatJson(item.params) || item.initialValue,
        type: 1
    };
};

// 撤回参数
const getRevokeOptions = (item, statusParam) => {
    return {
        ...item,
        url: item.url,
        disabled: !isDisabled(statusParam),
        onSuccess: item.onSuccess,
        initialValue: formatJson(item.params) || item.initialValue,
        type: 2
    };
};

//文件导入
const getFileImportOptions = (item, statusParam)=>{
    return {
        title:"导入",
        url: item.url,
        disabled: isDisabled(statusParam),
        onSuccess: item.onSuccess,
        fields: item.fields,
        ...item,
        style:item.style || item.setButtonStyle || {marginRight:5},
    };
}

// 文件自动导入
const getAutoFileImportOptions =  (item, statusParam)=>{
    return {
        url: item.url,
        fetchTable_1_Data: item.onSuccess,
        ...item
    };
}

//撤销导入
const getRevokeImportOptions =(item, statusParam)=>{
    return {
        ...item,
        title:"撤销导入",
        url: item.url,
        disabled: isDisabled(statusParam),
        onSuccess: item.onSuccess,
        fields: item.fields,
        initialValue: formatJson(item.params) || item.initialValue,
        style:item.style || item.setButtonStyle || {marginRight:5},
    };
}

//文件导出
const getFileExportOptions = (item, statusParam)=>{
    return {
        //disabled: false,
        title:"下载导入模板",
        url: item.url,
        onSuccess: item.onSuccess,
        ...item,
        disabled: (statusParam && isDisabled(statusParam)) || false,
        setButtonStyle:item.style || item.setButtonStyle || {marginRight:5},
    };
};

//下载导入模板
const getFileDownloadOptions = (item, statusParam) => {
    return {
        title: "下载导入模板",
        ...item,
        setButtonStyle: item.style || item.setButtonStyle || { marginRight: 5 },
        disabled: isDisabled(statusParam)
    }
}

//标记
const getMarkOptions = (item,statusParam) =>{

    /*if(isDisabled(statusParam)){
     item.formOptions.disabled = isDisabled(statusParam)
     }else{
     item.formOptions.disabled = !item.formOptions.selectedRowKeys.length>0
     }*/
    return {
        ...item,
        formOptions:{
            ...item.formOptions,
            disabled: isDisabled(statusParam) || (item.formOptions.selectedRowKeys && !item.formOptions.selectedRowKeys.length>0),
        },
        style:item.style || item.setButtonStyle || {marginRight:5},
    };
}

// form with modal
const getModalFormOptions = (item,statusParam) =>{
    return {
        ...item,
        buttonOptions:{
            icon:item.icon,
            text:item.title,
            style:item.style || item.setButtonStyle || {marginRight:5},
        },
        modalOptions:{
            title:item.title,
        },
        formOptions:{
            ...item.formOptions,
            disabled: isDisabled(statusParam),
            url:item.url,
            type:item.requestType || 'post',
            fields:item.fields,
            onSuccess:item.onSuccess,
        }
    };
}

// 数据匹配参数
/*const getMatchOptions = (item,statusParam) =>{
    return {
        ...item,
        url:item.url,
        filters:formatJson(item.params),
        disabled: isDisabled(statusParam),
        onSuccess: item.onSuccess,
        text:'数据匹配',
        icon:'copy',
        style: { marginRight: 5 }
    };
}*/

//table表格里面的操作 action
const getActionOptions = (item)=>{
    return {
        ...item,
        title:item.title,
        icon: item.icon,
        onSuccess: item.onSuccess,
        style:item.style || item.setButtonStyle || {marginRight:5},
    };
};

//switch
const getSwitchOptions = (item) =>{
    return {
        ...item,
        checked:item.checked,
        onSuccess: item.onSuccess,
        style:item.style || item.setButtonStyle || {marginRight:5},
    };
}

//buttons 参数形式
// [{type:'re',url:'',params:'',buttonOptions,PermissibleRender}]
const composeBotton = (buttons = [], params) => {
    return buttons.map((item, i) => {
        let component = undefined;
        if(item.type === 'add' || item.type === 'save' || item.type ==='view' || item.type === 'cancel' || item.type === 'edit' || item.type === 'delete' || item.type === 'retweet' || item.type === 'match' ){
            item.type = 'consistent'
        }

        switch (item.type) {
            case "consistent":
                component = (
                    <ButtonConsistent {...getConsistentOptions(item, params)} />
                );
                break;
            case "reset":
                component = (
                    <ButtonReset {...getResetOptions(item, params)} />
                );
                break;
            case "submit":
                component = (
                    <SubmitOrRecall
                        type={1}
                        {...getSubmitOptions(item, params)}
                    />
                );
                break;
            case "revoke":
                component = (
                    <SubmitOrRecall
                        type={2}
                        {...getRevokeOptions(item, params)}
                    />
                );
                break;
            case "fileImport":
                component = (
                    <FileImportModal
                        {...getFileImportOptions(item, params)}
                    />
                );
                break;
            case "autoFileImport":
                component = (
                    <AutoFileUpload
                        {...getAutoFileImportOptions(item, params)}
                    />
                );
                break;
            case "fileExport":
                component = (
                    <FileExport {...getFileExportOptions(item,params)} />
                );
                break;
            case "fileDownload":
                component = (
                    <FileDownload { ...getFileDownloadOptions(item) } />
                )
                break
            case 'revokeImport':
                component = (
                    <FileUndoImportModal {...getRevokeImportOptions(item,params)} />
                )
                break;
            case "mark":
                component = (
                    <ButtonMarkModal {...getMarkOptions(item, params)} />
                );
                break;
            case "modal":
                component =(
                    <ButtonModalWithForm {...getModalFormOptions(item,params)}/>
                )
                break;            
            /*case 'match':
                component = (
                    <ButtonReset {...getMatchOptions(item, params)} />
                );
                break;*/
            case 'action':
                component = (
                    <ButtonTableAction {...getActionOptions(item)} />
                );
                break;
            case 'switch':
                component = (
                    <ButtonSwitch {...getSwitchOptions(item)} />
                )
                break;
            case 'self':
                component = item.component
                break;
            default:
                break;
        }
        return typeof (item.userPermissions) !== 'undefined' ? (
            component && (
                <PermissibleRender
                    key={i}
                    userPermissions={item.userPermissions || []}>
                    {component}
                </PermissibleRender>
            )
        ) : <span key={i}>{component}</span>

    });
};

export default composeBotton;
