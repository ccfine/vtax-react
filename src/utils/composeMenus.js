/**
 * author       : liuliyuan
 * createTime   : 2017/12/7 16:05
 * description  :
 */

/*export default function composeMenus(nodeList) {
    const arr = [];
    nodeList.forEach((node) => {
        const item = node;
        item.exact = true;
        if (item.children && !item.component) {
            arr.push(...composeMenus(item.children));
        } else {
            if (item.children && item.component) {
                item.exact = false;
            }
            arr.push(item);
        }
    });

    return arr;
}*/





export default function composeMenus(nodeList) {
    const arr = [];
    nodeList.forEach((item) => {
        if (item.children) {
            arr.push({...item}, ...composeMenus(item.children));
        }else{
            arr.push(item);
        }
    });

    return arr;
}

