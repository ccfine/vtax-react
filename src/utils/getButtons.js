/*
 * @Author: liuchunxiu 
 * @Date: 2018-05-16 14:51:15 
 * @Last Modified by: liuchunxiu
 * @Last Modified time: 2018-05-18 10:20:39
 */
import React from "react";
import ButtonWithPut from "../compoments/buttonWithPut";
import SubmitOrRecall from "compoments/buttonModalWithForm/SubmitOrRecall.r";
import PermissibleRender from "compoments/permissible/PermissibleRender.r";
import { FileExport, FileImportModal } from "compoments";
//1：暂存 2：提交
// 判断当前状态是否提交
const isSubmit = (statusParam = {}) => {
    return parseInt(statusParam.status, 10) === 2;
};

// 判断参数数据是否存在
const hasParams = (params = {}) => {
    let hasMainId = false,
        hasMonth = false;
    for (let index in params) {
        if (/^mainId$/i.test(index) && params[index]) {
            hasMainId = true;
        } else if (/month/i.test(index) && params[index]) {
            hasMonth = true;
        }
    }

    return hasMainId && hasMonth;
};

// 重算参数
const getRecaculateOptions = (item, statusParam) => {
    // url,参数不存在，或是已经提交 重算不可用
    let res = { ...item, type: undefined };
    res.buttonOptions = res.buttonOptions || {};
    res.buttonOptions.disabled = true;
    res.buttonOptions.style = { marginRight: 5 };
    if (
        res.url &&
        res.params &&
        hasParams(res.params) &&
        !isSubmit(statusParam)
    ) {
        res.buttonOptions.disabled = false;
    }
    return res;
};

// 提交参数
const getSubmitOptions = (item, statusParam) => {
    let res = {
        ...item,
        url: item.url,
        disabled: true,
        onSuccess: item.onSuccess,
        initialValue: item.params || item.initialValue,
        type: 1
    };
    if (
        res.url &&
        res.initialValue &&
        hasParams(res.initialValue) &&
        !isSubmit(statusParam)
    ) {
        res.disabled = false;
    }
    return res;
};

// 撤回参数
const getRecallOptions = (item, statusParam) => {
    let res = {
        ...item,
        url: item.url,
        disabled: true,
        onSuccess: item.onSuccess,
        initialValue: item.params || item.initialValue,
        type: 2
    };
    if (
        res.url &&
        res.initialValue &&
        hasParams(res.initialValue) &&
        isSubmit(statusParam)
    ) {
        res.disabled = false;
    }
    return res;
};

//文件导入
const getFileImportOptions = (item, statusParam) => {
    let res = {
        title: "导入",
        url: item.url,
        disabled: false,
        onSuccess: item.onSuccess,
        fields: item.fields,
        ...item,
        style: item.style || item.setButtonStyle || { marginRight: 5 }
    };
    return res;
};

//文件导出
const getFileExportOptions = (item, statusParam) => {
    let res = {
        disabled: false,
        title: "下载导入模板",
        url: item.url,
        onSuccess: item.onSuccess,
        ...item,
        setButtonStyle: item.style || item.setButtonStyle || { marginRight: 5 }
    };
    return res;
};

//buttons 参数形式
// [{type:'re',url:'',params:'',buttonOptions,PermissibleRender}]
const getButtons = (buttons = [], ...params) => {
    let buttonElements = buttons.map(item => {
        let component = undefined;
        switch (item.type) {
            case "recaculate":
                component = (
                    <ButtonWithPut {...getRecaculateOptions(item, ...params)} />
                );
                break;
            case "submit":
                component = (
                    <SubmitOrRecall
                        type={1}
                        {...getSubmitOptions(item, ...params)}
                    />
                );
                break;
            case "recall":
                component = (
                    <SubmitOrRecall
                        type={2}
                        {...getRecallOptions(item, ...params)}
                    />
                );
                break;
            case "fileImport":
                component = (
                    <FileImportModal
                        {...getFileImportOptions(item, ...params)}
                    />
                );
                break;
            case "fileExport":
                component = (
                    <FileExport {...getFileExportOptions(item, ...params)} />
                );
                break;
            default:
                break;
        }

        return (
            component && (
                <PermissibleRender
                    key={item.key || item.type}
                    userPermissions={item.userPermissions || []}>
                    {component}
                </PermissibleRender>
            )
        );
    });

    return buttonElements;
};

export default getButtons;
