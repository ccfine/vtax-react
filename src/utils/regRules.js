/**
 * Created by liuliyuan on 2017/9/27.
 */

export default {
    userName:{
        pattern:/^[a-zA-Z0-9]{6,20}?$/,
        message:'请输入有效的用户名'
    },
    password:{
        pattern:/(?![0-9]+$)(?![a-zA-Z]+$)(?!(![0-9A-Za-z])+$)\S{6,32}/,
        message:'请输入6-32位由数字与字母组合的密码'
    },
    companyPhone:{
        pattern:/(^\d{3,4}-?\d{7,8}$)|(^1\d{10}$)/,
        message:'请输入正确的联系电话'
    },
    trim:{
        pattern:/^\S+(\s+\S+)*$/,
        message:'前后不能包含空格'
    },
    email:{
        pattern:/^([_a-z0-9]|[.]|[-])+@(([_a-z0-9]|[-])+\.)+[a-z0-9]+$/i,
        message:'非法电子邮件地址'
    },
    postcode:{
        pattern:/^(\d{6})$/,
        message:'非法邮政编码'
    },
    mobile_phone:{
        pattern:/^1[3578](\d{9})$/,
        message:'非法手机号码'
    },
    phone_with_area_code:{
        pattern:/^[0-9]{3,4}-[0-9]{3,11}(-[0-9]{0,}){0,1}$/,
        message:'号码中区号应为3位或4位数字，之后为分隔线-，电话号码为3-11位的数字，最后可跟分机号，电话号码和分机号之间也需要有-分隔，如021-50801155-1004'
    },
    phone_without_area_code:{
        pattern:/^[1-9]{1}[0-9]{2,10}$/,
        message:'电话号码应为3-11位的数字，需以非零开头'
    },
    integer:{
        pattern:/^[-]{0,1}[1-9]+[0-9]{0,}$|^[0]{1,1}$/,
        message:'必须输入整数'
    },
    positive_integer:{
        pattern:/^[1-9]+[0-9]{0,}$/,
        message:'必须输入正整数'
    },
    nonnegative_integer:{
        pattern:/^[1-9]+[0-9]{0,}$|^[0]{1,1}$/,
        message:'必须输入非负整数'
    },
    negative_integer:{
        pattern:/^-[1-9]+[0-9]{0,}$/,
        message:'必须输入负整数'
    },
    http_url:{
        pattern:/^https?:\/\/(([a-zA-Z0-9_-])+(\.)?)*(:\d+)?(\/((\.)?(\?)?=?&?[a-zA-Z0-9_-](\?)?)*)*$/i,
        message:'需要以http://或https://开头'
    },
    number:{
        pattern:/^[-]{0,1}[1-9]+[0-9]{0,}$|^[-]{0,1}[1-9]+[0-9]{0,}[.][0-9]+$|^[-]{0,1}[0]{0,1}[.][0-9]+$|^[0]{1,1}$/,
        message:'必须输入数字'
    },
    nonnegative_number:{
        pattern:/^[1-9]+[0-9]{0,}$|^[1-9]+[0-9]{0,}[.][0-9]+$|^[0]{0,1}[.][0-9]+$|^[0]{1,1}$/,
        message:'必须输入非负数字'
    },
    positive_number:{
        pattern:/^[1-9]+[0-9]{0,}$|^[1-9]+[0-9]{0,}[.][0-9]+$|^[0]{0,1}[\][0-9]+$/,
        message:'必须输入正数字'
    },
    decimal:{
        pattern:/^[-]{0,1}[0]{0,1}[.][0-9]+$/,
        message:'必须输入小数'
    },
    label:{
        pattern:/^[a-z]{1}([a-z0-9]|[_])+$/i,
        message:'仅包含字母数字和下划线，需要以字母开头'
    },
    string:{
        pattern:/^([a-z0-9]|[_])+$/i,
        message:'英文字符串必须仅由字母、数字和下划线组成'
    },
    chinese_string:{
        pattern:/^([a-z0-9]|[_]|[\u4E00-\u9FA5])+$/i,
        message:'中文字符串仅包含中文，英文字母数字和下划线'
    },
    not_chinese_string:{
        pattern:/^[^\u4E00-\u9FA5]*$/i,
        message:'包含非法的中文字符串'
    },
    not_chinese:{
        pattern:/^[^\u4e00-\u9fa5]{0,}$/,
        message:'不能输入汉字'
    },
    ip_address:{
        pattern:/^([0-9]|[1-9][0-9]|[1][0-9]{2}|[2][0-4][0-9]|25[0-5])([.]([0-9]|[1-9][0-9]|[1][0-9]{2}|[2][0-4][0-9]|25[0-5])){3}$/,
        message:'非法的IP地址'
    },
    text:{
        pattern:/^([\w]|[\W])*$/i,
        message:'非法的字符串'
    },
    mac:{
        pattern:/^[0-9a-f]{2}([:][0-9a-f]{2}){5}$|^[0-9a-f]{2}([-][0-9a-f]{2}){5}$/i,
        message:'非法的MAC地址'
    },
    textarea_length_100:{
        max:100,
        message:'不能超过100个字符'
    },
    textArea_length_225:{
        max:255,
        message:'不能超过255个字符'
    },
    textarea_length_2000:{
        max:2000,
        message:'不能超过2000个字符'
    },
    input_length_20:{
        max:20,
        message:'不能超过20个字符'
    },
    input_length_25:{
        max:25,
        message:'不能超过25个字符'
    },
    input_length_50:{
        max:50,
        message:'不能超过50个字符'
    }
}