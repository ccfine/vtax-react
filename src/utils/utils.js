/**
 * author       : liuliyuan
 * createTime   : 2017/12/16 9:45
 * description  :
 */

const fMoney = (s,n=2)=>{

    if(s === "" || s === 0 || typeof (s) === 'undefined'){
        return '0.00';
    }else{
        n = n > 0 && n <= 20 ? n : 2;
        s = Math.floor(parseFloat((s + "").replace(/[^\d\\.-]/g, "")) * 100 ) /100 + "";
        let l = s.split(".")[0].split("").reverse(),
            r = s.split(".")[1] || 0;
        if(r===0 || r.length ===1){
            r+='0';
        }
        let t = "";
        for(let i = 0; i < l.length; i ++ )
        {
            t += l[i] + ((i + 1) % 3 === 0 && (i + 1) !== l.length ? "," : "");
        }
        return t.split("").reverse().join("") + "." + r;
    }
}

const digitUppercase=(n)=> {
    const fraction = ['角', '分'];
    const digit = ['零', '壹', '贰', '叁', '肆', '伍', '陆', '柒', '捌', '玖'];
    const unit = [
        ['元', '万', '亿'],
        ['', '拾', '佰', '仟'],
    ];
    let num = Math.abs(n);
    let s = '';
    fraction.forEach((item, index) => {
        s += (digit[Math.floor(num * 10 * (10 ** index)) % 10] + item).replace(/零./, '');
    });
    s = s || '整';
    num = Math.floor(num);
    for (let i = 0; i < unit[0].length && num > 0; i += 1) {
        let p = '';
        for (let j = 0; j < unit[1].length && num > 0; j += 1) {
            p = digit[num % 10] + unit[1][j] + p;
            num = Math.floor(num / 10);
        }
        s = p.replace(/(零.)*零$/, '').replace(/^$/, '零') + unit[0][i] + s;
    }

    return s.replace(/(零.)*零元/, '元').replace(/(零.)+/g, '零').replace(/^整$/, '零元整');
}

//数组删除的方法
Array.prototype.remove = function(index){
    if(isNaN(index) || index > this.length){
        return false;
    }
    for(let i=0,n=0;i<this.length;i++){
        if(this[i] !== this[index]){
            this[n++] = this[i];
        }
    }
    this.length -= 1;
}


export {
    fMoney,
    digitUppercase
}