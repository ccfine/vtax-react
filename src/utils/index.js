/**
 * author       : liuliyuan
 * createTime   : 2017/12/5 18:09
 * description  :
 */
import React from 'react';
import request from './request'
import composeMenus from './composeMenus'
import regRules from './regRules'
import getFields from './getFields'
import {BigNumber} from 'bignumber.js'

const fMoney = (s,n=2)=>{
    if(s === "" || s === 0 || typeof (s) === 'undefined'){
        return '0.00';
    }
    n = n > 0 && n <= 20 ? n : 2;
    /**添加一下代码 大数字用parseFloat不精确 */
    s = s.toString().replace(/[^\d\\.-]/g, "");
    return (new BigNumber(s)).toFormat(n);

    // s = parseFloat((s + "").replace(/[^\d\\.-]/g, "")).toFixed(n) + "";
    // /**过滤负号 .replace(/-/g,'') 解决负数转换 2018/3/19*/
    // let l = s.split(".")[0].replace(/-/g,'').split("").reverse(),
    //     r = s.split(".")[1];
    // let t = "";
    // for (let i = 0; i < l.length; i++) {
    //     t += l[i] + ((i + 1) % 3 === 0 && (i + 1) !== l.length ? "," : "");
    // }
    // /**添加负号 (s.indexOf('-')>-1?"-":"") + 解决负数转换 2018/3/19*/
    // return  (s.indexOf('-')>-1?"-":"") + t.split("").reverse().join("") + "." + r;
}
const getUrlParam = function(name) {
    var reg = new RegExp("(^|&)" + name + "=([^&]*)(&|$)"); //构造一个含有目标参数的正则表达式对象
    var r = window.location.search.substr(1).match(reg);  //匹配目标参数
    if (r != null) return decodeURI(r[2]); return null; //返回参数值
}
const getDict = type => {
    return new Promise(function (resolve, reject) {
        request.get(`/sys/dict/listBaseInfo/${type}`)
            .then(({data})=>{
                if(data.code===200){
                    resolve(data.data)
                }else{
                    reject(data.msg)
                }
            })
    })
}
const requestDict = async (type,callback)=>{
    let result = await getDict(type);
    callback(result)
}

/**
 ** 除法函数，用来得到精确的除法结果
 ** 说明：javascript的除法结果会有误差，在两个浮点数相除的时候会比较明显。这个函数返回较为精确的除法结果。
 ** 调用：accDiv(arg1,arg2)
 ** 返回值：arg1除以arg2的精确结果
 **/
const accDiv=(arg1, arg2)=> {
    if(parseInt(arg1, 0) === 0 && parseInt(arg2, 0) === 0) return 0;

    let t1 = 0, t2 = 0, r1, r2;
    try {
        t1 = arg1.toString().split(".")[1].length;
    }
    catch (e) {
    }
    try {
        t2 = arg2.toString().split(".")[1].length;
    }
    catch (e) {
    }

    r1 = Number(arg1.toString().replace(".", ""));
    r2 = Number(arg2.toString().replace(".", ""));
    return (r1 / r2) * Math.pow(10, t2 - t1);

}

/**
 ** 乘法函数，用来得到精确的乘法结果
 ** 说明：javascript的乘法结果会有误差，在两个浮点数相乘的时候会比较明显。这个函数返回较为精确的乘法结果。
 ** 调用：accMul(arg1,arg2)
 ** 返回值：arg1乘以 arg2的精确结果
 **/
const accMul=(arg1, arg2)=> {
    let m = 0, s1 = arg1.toString(), s2 = arg2.toString();
    try {
        m += s1.split(".")[1].length;
    }
    catch (e) {
    }
    try {
        m += s2.split(".")[1].length;
    }
    catch (e) {
    }
    return Number(s1.replace(".", "")) * Number(s2.replace(".", "")) / Math.pow(10, m);
}

/**
 ** 减法函数，用来得到精确的减法结果
 ** 说明：javascript的减法结果会有误差，在两个浮点数相减的时候会比较明显。这个函数返回较为精确的减法结果。
 ** 调用：accSub(arg1,arg2)
 ** 返回值：arg1加上arg2的精确结果
 **/
const accSub=(arg1, arg2)=> {
    let r1, r2, m, n;
    try {
        r1 = arg1.toString().split(".")[1].length;
    }
    catch (e) {
        r1 = 0;
    }
    try {
        r2 = arg2.toString().split(".")[1].length;
    }
    catch (e) {
        r2 = 0;
    }
    m = Math.pow(10, Math.max(r1, r2)); //last modify by deeka //动态控制精度长度
    n = (r1 >= r2) ? r1 : r2;
    return ((arg1 * m - arg2 * m) / m).toFixed(n);
};

/**
 ** 加法函数，用来得到精确的加法结果
 ** 说明：javascript的加法结果会有误差，在两个浮点数相加的时候会比较明显。这个函数返回较为精确的加法结果。
 ** 调用：accAdd(arg1,arg2)
 ** 返回值：arg1加上arg2的精确结果
 **/
const accAdd=(arg1, arg2)=>{
    let r1, r2, m, c;
    try {
        r1 = arg1.toString().split(".")[1].length;
    }
    catch (e) {
        r1 = 0;
    }
    try {
        r2 = arg2.toString().split(".")[1].length;
    }
    catch (e) {
        r2 = 0;
    }
    c = Math.abs(r1 - r2);
    m = Math.pow(10, Math.max(r1, r2));
    if (c > 0) {
        let cm = Math.pow(10, c);
        if (r1 > r2) {
            arg1 = Number(arg1.toString().replace(".", ""));
            arg2 = Number(arg2.toString().replace(".", "")) * cm;
        } else {
            arg1 = Number(arg1.toString().replace(".", "")) * cm;
            arg2 = Number(arg2.toString().replace(".", ""));
        }
    } else {
        arg1 = Number(arg1.toString().replace(".", ""));
        arg2 = Number(arg2.toString().replace(".", ""));
    }
    return (arg1 + arg2) / m;
};

//获取html
const htmlDecode = html =>{
    if(html){
        let div = document.createElement( 'div' );
        div.innerHTML = html;
        return div.textContent;
    }
};

//将0.5转换成50%
const toPercent = val=>{
    let valNum = Number(val);
    if(isNaN(valNum) || valNum === 0)return val;
    return `${valNum*100}%`;
}

//将50%转换成0.5
const fromPercent = val=>{
    let valTrim = val.replace?val.replace('%',''):val;
    let valNum = Number(valTrim);
    if(isNaN(valNum))return val;
    return valNum/100;
}

// 提交、撤回相关 状态显示
const transformDataStatus = status =>{
    status = parseInt(status,0)
    if(status===1){
        return '暂存';
    }
    if(status===2){
        return '提交'
    }
    return status
}
const listMainResultStatus = (statusParam) =>{
    return (statusParam && statusParam.status) && <div style={{marginRight: 30, display: 'inline-block'}}>
                                <span style={{marginRight: 20}}>状态：<label style={{color: '#f5222d'}}>{
                                    transformDataStatus(statusParam.status)
                                }</label></span>
        {
            statusParam.lastModifiedDate && <span>提交时间：{statusParam.lastModifiedDate}</span>
        }
    </div>
}

export {
    regRules,
    request,
    fMoney,
    getUrlParam,
    composeMenus,
    requestDict,
    accDiv,
    accMul,
    accSub,
    accAdd,
    getFields,
    htmlDecode,
    toPercent,
    fromPercent,
    listMainResultStatus,
}