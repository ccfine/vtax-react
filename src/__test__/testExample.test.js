/**
 * Created by liuliyuan on 2018/4/24.
 * 测试示例
 */
import React from 'react'
import {findDOMNode} from 'react-dom';
import {withRouter} from 'react-router-dom';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { accDiv } from 'utils'
import Message from 'modules/header/Message.react'

Enzyme.configure({ adapter: new Adapter() });

describe('测试', function () {
    it('测试除法函数', function () {
        expect(accDiv(100,100)).toEqual(1)
    });

    it('测试组件加载', function () {
        expect(shallow(<Message />)).toMatchSnapshot();
    });

});

describe('测试2', function () {

    let Messages = withRouter(Message);

    /*
     * 之后，我们可以:
     * 通过wrapper.state()拿到组件的state
     * 通过wrapper.instance()拿到组件实例，以此调用组件内的方法
     * 通过wrapper.find()找到组件内的子组件
     * 但是，无法通过wrapper.props()拿到组件的props
     */

    it("测试组件加载-1", function() {
        expect(shallow(<Messages />).contains(<div className="action" />)).toBe(true);
    });

    //测试该组件组外层的class
    it("测试组件加载-2", function() {
        //expect(shallow(<Messages />).is('.action')).toEqual(true);
        expect(shallow(<Messages />).is('.action')).toBe(true);
    });

    it("测试组件加载-3", function() {
        expect(shallow(<Messages />).find('.action').length).toBe(1);
    });

});



describe("测试2", function() {

    const Foo = () =><div className="foo" />;

    it("测试组件加载2-1", function() {
        expect(shallow(<Foo />).contains(<div className="foo" />)).toBe(true);
    });

    it("测试组件加载2-2", function() {
        expect(shallow(<Foo />).is('.foo')).toBe(true);
    });

    it("测试组件加载2-3", function() {
        expect(mount(<Foo />).find('.foo').length).toBe(1);
    });
});
