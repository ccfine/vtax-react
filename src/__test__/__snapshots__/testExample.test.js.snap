// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 ImageGallery 组件 测试组件加载 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Message) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`测试 测试组件加载 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(Message) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
